python/vulns_per_assets/export_asset_vulns.py

```diff
@@ -0,0 +1,241 @@
+        secs += wait_interval_secs 
+
+    print("")
+    if secs >= wait_limit_secs:
+        print(f"Waited for {wait_limit_secs} seconds.")
+        print(f"Consider re-running with search ID")
+        sys.exit(1)
+ 
+# Obtain the exported asset data.
+def retrieve_asset_data(api_key, base_url, id, asset_file_name):
+    get_data_url = base_url + "/data_exports/?search_id=" + id
+    headers = {'Accept': 'application/gzip; charset=utf-8',
+               'X-Risk-Token': api_key}
+    
+    gz_asset_file_name = asset_file_name + ".gz"
+    try:
+        response = requests.get(get_data_url, headers=headers, stream=True)
+    
+        with open(gz_asset_file_name, 'wb') as file_gz:
+            for block in response.iter_content(8192):
+                file_gz.write(block)
+    
+    except Exception as exp:
+        print(f"Retrieve asset data error: {exp.__str__()}")
+        sys.exit(1)
+
+    return gz_asset_file_name
+
+# Count the number of lines in the unzip asset file.
+# From: https://stackoverflow.com/questions/845058/how-to-get-line-count-of-a-large-file-cheaply-in-python
+def count_lines(asset_file_name):
+    f = open(asset_file_name, 'rb')
+    lines = 0
+    buf_size = 1024 * 1024
+    read_f = f.raw.read
+
+    buf = read_f(buf_size)
+    while buf:
+        lines += buf.count(b'\n')
+        buf = read_f(buf_size)
+    
+    f.close()
+
+    # Add one to the number of lines because the end of files does not have '\n'.
+    return lines + 1
+
+# Obtain and write vulnerability information per asset ID into the specified file.
+def get_vuln_info(api_key, vuln_url, asset_id, avfp, avlfp):
+    headers = {'Accept': 'application/json',
+               'Content-Type': 'application/json; charset=utf-8',
+               'X-Risk-Token': api_key}
+
+    vuln_url = "https://" + vuln_url
+
+    retry_cnt = 0
+    success = False
+    while not success:
+       try:
+           response = requests.get(vuln_url, headers=headers)
+           http_status_code = response.status_code
+   
+           # If too many requests, wait a second.  If is happens again, error out.
+           if http_status_code == 429:
+               time.sleep(1)
+               response = requests.get(vuln_url, headers=headers)
+               response.raise_for_status()
+   
+       except requests.Timeout as tme:
+           retry_cnt += 1
+           print(f"\nGet vuln info Timeout error: {tme.__str__()}.  Sleeping 60s ({retry_cnt})")
+           if retry_cnt > 3:
+               return 
+           time.sleep(60)
+   
+       except Exception as exp:
+           print(f"\nGet vuln info error: {exp.__str__()}")
+           return
+        
+       success = True
+    
+    resp_json = response.json()
+    
+    vulns = resp_json['vulnerabilities']
+    num_vulns = len(vulns)
+    print(f"Vulnerabilities for asset ID {asset_id} ({num_vulns})", file=avlfp)
+    for vuln in vulns:
+        print(vuln, file=avfp)
+
+    return num_vulns
+
+# main
```

Comment:
consider wrapping all of the following code under `if __name__ == "__main__":` This will cause the code following that if statement to only run if this file is used as a script. That means that the functions defined in this file can be imported into other scripts and programs without inadvertently calling this main block.

Reply:
Done.